### PR TITLE
WAITP-1391 LESMIS to vite

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- Converted LESMIS project to `vite`, including renaming jsx files to `.jsx` extension and updating build scripts
Note: Had to set `NODE_OPTIONS=--max-old-space-size=8192` on build because of a known issue with Vite and large apps

### CODE REVIEW

File: packages/lesmis/package.json [hunk 1]
```diff
@@ -10,23 +10,21 @@
   },
   "author": "Beyond Essential Systems <admin@tupaia.org> (https://beyondessential.com.au)",
   "main": "dist/index.js",
-  "scripts": {
-    "build": "npm run --prefix ../../ package:build:react-scripts",
+  "scripts": { 
+    "build": "cross-env NODE_OPTIONS=--max-old-space-size=8192 npm run --prefix ../../ package:build:vite",
```
Comment (acdunham) [inline packages/lesmis/package.json:14]: This is a workout for a [known issue with Vite](https://github.com/vitejs/vite/issues/2433)

Comment (tcaiger) [reply packages/lesmis/package.json:14]: A bit of a concern but well found

File: packages/lesmis/src/main.jsx [hunk 2]
```diff
@@ -22,10 +22,11 @@ const render = () => {
...
-if (module.hot) {
-  module.hot.accept('./App', () => {
+if (import.meta.hot) {
+  import.meta.hot.accept('./App', () => {
     // eslint-disable-next-line global-require
     const NextApp = require('./App').default;
     render(NextApp);
   });
 }
+
```
Comment (tcaiger) [inline packages/lesmis/src/main.jsx:32]:
```suggestion

```
Actually can we just remove this code. It is just a work around for the fact that react-scripts hot reload didn't work properly.

**Review by tcaiger — APPROVED** [review]
Great to have this done. Just left a comment about a tiny clean up of some legacy hot reload code. thanks

**Pull request merged**